crates/nu-command/src/env/config/utils.rs

```diff
@@ -31,7 +31,7 @@ pub(crate) fn gen_command(
     ExternalCommand {
         name,
         args,
-        arg_keep_raw: vec![false; number_of_args],
+        arg_constraints: vec![(false, false); number_of_args],
```

Comment:
Can you add a comment that explains what the two booleans mean? This is a problem in general that in many places in our code we have a `bool` somewhere without annotating what it means. It could also be done by adding another array to `ExternalCommand` named `arg_is_glob` or similar, instead of having the tuples.

Reply:
Based on our discussion yesterday, I think we need to disable the implicit cast from `Glob` to `String`.

So I revert my previous pr, and something like `let f: glob = "*.nu"; ^echo $f` is no longer supported.